### PR TITLE
Make layer methods public

### PR DESCRIPTION
**Pull request by clienthax**
Makes it possible to add extra layers to the renderer for players without having to use reflection/mod specific AT's

### CODE REVIEW

File: src/main/resources/forge_at.cfg [hunk 1]
```diff
@@ -157,6 +157,11 @@ public net.minecraft.client.renderer.WorldRenderer func_78909_a(I)I # getColorIn
 public net.minecraft.client.renderer.WorldRenderer func_178972_a(IIIII)V # putColorRGBA
 # ModelBlock Constructor
 public net.minecraft.client.renderer.block.model.ModelBlock <init>(Lnet/minecraft/util/ResourceLocation;Ljava/util/List;Ljava/util/Map;ZZLnet/minecraft/client/renderer/block/model/ItemCameraTransforms;)V
+# RenderManager
+public net.minecraft.client.renderer.entity.RenderManager field_178636_l # skinMap
```
Comment (LexManos) [inline src/main/resources/forge_at.cfg]: Why, this has nothing to do with layers. No

File: src/main/resources/forge_at.cfg [hunk 2]
```diff
@@ -157,6 +157,11 @@ public net.minecraft.client.renderer.WorldRenderer func_78909_a(I)I # getColorIn
...
+# RenderLivingEntity
+public net.minecraft.client.renderer.entity.RendererLivingEntity func_177094_a(Lnet/minecraft/client/renderer/entity/layers/LayerRenderer;)Z # addLayer
+public net.minecraft.client.renderer.entity.RendererLivingEntity func_177089_b(Lnet/minecraft/client/renderer/entity/layers/LayerRenderer;)Z # removeLayer
```
Comment (LexManos) [inline src/main/resources/forge_at.cfg]: Adding, i can get behind that. But removing is a no-go. Encouraging others to fuck with other's additions is stupid.
If you dont want your additional item to be rendered, just don't do it in your handler.

File: src/main/resources/forge_at.cfg (hunk 1 above)
Comment (clienthax) [inline src/main/resources/forge_at.cfg]: because you need to get the render from the skinmap if you are adding a new generic layer (like for hats) ?

Comment (LexManos) [inline src/main/resources/forge_at.cfg]: Then add a getter, not a "let's let anyone replace this wholesale willy nilly without informing anyone else"

Comment (clienthax) [inline src/main/resources/forge_at.cfg]: Would a getter actualyl be better, you could just call clear on the map anyway.

Comment (diesieben07) [inline src/main/resources/forge_at.cfg]: `Collections.unmodifiableMap`

Comment (LexManos) [inline src/main/resources/forge_at.cfg]: Dont return the map, or use a unmodifable view.

**Comment by clienthax** [comment]
@LexManos look good?

**Comment by cpw** [comment]
Use an immutablemap, they have better performance..

**Comment by diesieben07** [comment]
But they would make a copy...

**Comment by cpw** [comment]
And unmodifiable map doesn't? They both make copies of the parent structure. Neither make deep clones..

**Comment by clienthax** [comment]
I'm not familiar with these maps, but it needs to be possible to call methods on the values that are in the skinmap

**Comment by diesieben07** [comment]
unmodifiableMap is a wrapper. ImmutableMap is a copy of the contents.

**Comment by LexManos** [comment]
Do you need to have a itteable of the listed keys? or is being able to request a specifc key good enough?
Also, when doing unmodifiable it's better to cache the result as it can be used as the result many times.

**Comment by diesieben07** [comment]
Since when do we do overoptimizations like caching those in forge?

**Comment by clienthax** [comment]
@LexManos it would be nice to at least use foreach, to add a layer to all current and possible future player skin types they may add

Atm we use something like this
`
		for(RenderPlayer renderPlayer : ((HashMap<String, RenderPlayer>) Minecraft.getMinecraft().getRenderManager().skinMap).values()) {
			renderPlayer.addLayer(new LayerHat(renderPlayer.getPlayerModel().bipedHead));
			renderPlayer.addLayer(new LayerSash(renderPlayer));
		}
`

**Comment by clienthax** [comment]
@LexManos Does this need any changes?

**Comment by iChun** [comment]
Heh, this'll save me a few ATs in my own mods in the future

**Pull request merged**